### DISCUSSION TITLE
EventHandler for Task inside Loop

### DISCUSSION THREAD

**Discussion by astelmashenko** [Netflix/conductor#2873] [Q&A] [answered]
There is a goal to do:

1. wait for user actions (cart submit)
2. once cart submitted, send event to start Payment processing
3. analyze Payment processing result
4. if success then End Workflow else run Loop from p.1 once again

So it is repetitive task, user work with a cart then tries to do payment, which may fail (not enough money), update cart and make successful payment which leads to order placed.
p.1 is just as WAIT task, which await for completion through it's EventHandler. There is a problem with EventHandler, it request taskRef. And in case of Loop task iterations is added to taskRef, so it will be myWaitTask__1, myWaitTask__2 and so on.

How can I create EventHandler for task when it's taskRef is changing?

**Comment by manan164** [comment, marked as answer]
Hi @astelmashenko , Currently it is not supported. We will add support for loop over tasks also. The implementation will be the same as other tasks, The task ref name should not be passed as appended iteration value. Thanks for the suggestion.

**Comment by astelmashenko** [comment]
@manan164 , thank you for answer. Is there a ticket for that or a roadmap point, so we can follow and see when it is ready?

**Reply by manan164** [reply]
Hi @astelmashenko , Please keep an eye on https://github.com/Netflix/conductor/pull/2883.  I will update once it is released.
